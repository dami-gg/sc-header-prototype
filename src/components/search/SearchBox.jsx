import React, {Component} from 'react';

import './Search.scss';

class SearchBox extends Component {
  render() {
    return (
        <input className="dc-input dc-search-form__input"
               onChange={(event) => {
                 return this.props.onChangeAction && this.props.onChangeAction(event)
               }}
               placeholder="Search..."
        />
    );
  }
}

SearchBox.propTypes = {
  onChangeAction: React.PropTypes.func
};

export default SearchBox;
