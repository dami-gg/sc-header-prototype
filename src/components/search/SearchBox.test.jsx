import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from './SearchBox';

describe('SearchBox component', () => {
  it('should render correctly', () => {
    let tree = renderer.create(<SearchBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});