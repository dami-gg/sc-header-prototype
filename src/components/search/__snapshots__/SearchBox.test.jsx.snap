exports[`SearchBox component should render correctly 1`] = `
<input
  className="dc-input dc-search-form__input"
  onChange={[Function onChange]}
  placeholder="Search..." />
`;
